src/cascadia/TerminalApp/Tab.cpp

```diff
@@ -153,25 +154,7 @@ void Tab::UpdateIcon(const winrt::hstring iconPath)
     _lastIconPath = iconPath;
 
     _tabViewItem.Dispatcher().RunAsync(CoreDispatcherPriority::Normal, [this]() {
-        Controls::IconSourceElement elem{};
-
-        if (!_lastIconPath.empty())
-        {
-            try
-            {
-                winrt::Windows::Foundation::Uri iconUri{ _lastIconPath };
-                Controls::BitmapIconSource iconSource;
-                // Make sure to set this to false, so we keep the RGB data of the
-                // image. Otherwise, the icon will be white for all the
-                // non-transparent pixels in the image.
-                iconSource.ShowAsMonochrome(false);
-                iconSource.UriSource(iconUri);
-                elem.IconSource(iconSource);
-            }
-            CATCH_LOG();
-        }
-
-        _tabViewItem.Icon(elem);
+        _tabViewItem.Icon(GetColoredIcon(_lastIconPath));

```

Comment:
This one lost its exception handler.. It also no longer checks whether the path was empty. Does this work right for transitioning from "an icon" to "no icon"?

Reply:
The exception handler went up into GetColoredIcon, but I'll double check the icon->null path